controllers/front/SmailyCartCron.php

```diff
@@ -45,107 +45,157 @@ public function init()
+            $cart_updated_time = strtotime($abandoned_cart['date_upd']);
+            $reminder_time = strtotime('+' . $delay . ' minutes', $cart_updated_time);
+            $current_time = strtotime(date('Y-m-d H:i') . ':00');
+            // Don't continue if cart delay time has not passed.
+            if ($current_time < $reminder_time) {
+                continue;
+            }
+
+            // Get cart by id.
+            $id_customer = (int) $abandoned_cart['id_customer'];
+            $id_cart = (int) $abandoned_cart['id_cart'];
+            $cart = new Cart($abandoned_cart['id_cart']);
+
+            // Get cart products.
+            $products = $cart->getProducts();
+            // Don't continue if no products in cart.
+            if (empty($products)) {
+                continue;
+            }
+
+            // Initialize Smaily query.
+            $adresses = array(
+                'email' => $abandoned_cart['email'],
+            );
+
+            if (in_array('first_name', $sync_fields)) {
+                $adresses['first_name'] = $abandoned_cart['firstname'];
+            }
+
+            if (in_array('last_name', $sync_fields)) {
+                $adresses['last_name'] = $abandoned_cart['lastname'];
+            }
+
+            // Populate abandoned cart with empty values for legacy api.
+            $fields_available = array(
+                'name',
+                'description',
+                'price',
+                'category',
+                'quantity',
+                'base_price',
+            );
+            foreach ($fields_available as $field) {
+                for ($i=1; $i<=10; $i++) {
+                    $adresses['product_' . $field . '_' . $i] = '';
+                }
+            }
+
+            // Collect products of abandoned cart.
+            $count = 1;
+            foreach ($products as $product) {
+                // Get only 10 products.
+                if ($count > 10) {
+                    $adresses['over_10_products'] = 'true';
+                    break;
+                }
+                // Standardize template parameters across integrations.
+                foreach ($sync_fields as $sync_field) {
+                    switch ($sync_field) {
+                        case 'first_name':
+                        case 'last_name':
+                            // Skip customer fields
+                            break;
+                        case 'base_price':
+                            $adresses['product_base_price_' . $count] = Tools::displayPrice(
+                                $product['price_without_reduction']
+                            );
+                            break;
+                        case 'price':
+                            $adresses['product_price_' . $count] = Tools::displayPrice(
+                                $product['price_with_reduction']
+                            );
+                            break;
```

Comment:
`base_price` and `price` fields now have also formating and pass values as displayed in the store. Same as Magento 2 plugin

Reply:
Awesome!